src/lib/StorageFileApi.ts

```diff
@@ -178,6 +178,29 @@ export class StorageFileApi {
     }
   }
 
+  /**
+   * Downloads a file.
+   *
+   * @param path The file path to be downloaded, including the path and file name. For example `folder/image.png`.
+   */
+  async getPublicUrl(
+    path: string
+  ): Promise<{
+    data: { publicURL: string } | null
+    error: Error | null
+    publicURL: string | null
+  }> {
+    try {
+      const _path = this._getFinalPath(path)
+      let publicURL = `/object/public/${_path}`
```

Comment:
I was thinking if we should add a check here to see if the bucket is public and the object exists at the specified path.

Reply:
Lets keep this method simple Tyler. Cos even if this check passes and we return the URL, by the time the user downloads the asset, the permissions might have changed or object might have been deleted.

Reply:
Sounds good @inian !